haproxy/state/from_ha.go

```diff
@@ -51,30 +51,44 @@ func FromHAProxy(ha HAProxyRead) (State, error) {
 		if err != nil {
 			return state, err
 		}
-		if len(logTargets) > 1 {
-			return state, fmt.Errorf("expected at most 1 filter for frontend %s, got %d", f.Name, len(filters))
+		if len(logTargets) >= 2 {
+			return state, fmt.Errorf("expected at most 2 filters for frontend %s, got %d", f.Name, len(filters))
 		}
-		var filter *FrontendFilter
-		if len(filters) == 1 {
-			filter = &FrontendFilter{
-				Filter: filters[0],
+		var filter_spoe *FrontendFilter
+		var filter_compression *FrontendFilter
+		if len(filters) <= 2 {
+			index_compression_filter := 0
+			index_spoe_filter := 1
+			if filters[0].Type == models.FilterTypeSpoe {
+				index_spoe_filter = 0
+				index_compression_filter = 1
 			}
-
-			rules, err := ha.TCPRequestRules("frontend", f.Name)
-			if err != nil {
-				return state, err
+			filter_compression = &FrontendFilter{
+				Filter: filters[index_compression_filter],
 			}
-			if len(binds) != 1 {
-				return state, fmt.Errorf("expected 1 tcp request rule for frontend %s, got %d", f.Name, len(rules))
+
+			if len(filters) == 2 {
+				filter_spoe = &FrontendFilter{
```

Comment:
How can you be sure SPOE filter is the second one? What about testing its type instead?

Reply:
Done